zerver/lib/actions.py

```diff
@@ -4691,6 +4691,12 @@ def do_change_tos_version(user_profile: UserProfile, tos_version: str) -> None:
     )
 
 
+def bulk_regenerate_api_keys(user_profile_ids: List[int]) -> None:
+    for user_profile_id in user_profile_ids:
+        user_profile = get_user_profile_by_id(user_profile_id)
+        do_regenerate_api_key(user_profile, user_profile)
```

Comment:
This isn't very efficient, but  kind of mimics how this will happen in parallel in session deletions, which also do `o(n)` operations

Reply:
Yeah, this is probably simplest given the `deferred_work` structure we have for removing push device tokens.